### PR TITLE
CodeCamp #141 [Feature] Add BioMedical3DRandomFlip.

### PR DESCRIPTION
**Pull request by Leeinsn**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support for biomedical 3d images augmentation.

## Modification

Add BioMedical3DRandomFlip in mmseg/datasets/transforms/transforms.py.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

Add the function to support the flipping for biomedical 3d images and segmentation maps.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by Leeinsn** [comment]
Yes, I check the information about errors, but I cannot find the reason for the bug of my code.

**Comment by MeowZheng** [comment]
> Yes, I check the information about errors, but I cannot find the reason for the bug of my code.

we are figuring out the bug of the ci test

File: mmseg/datasets/transforms/transforms.py [hunk 1]
```diff
@@ -1805,3 +1805,114 @@ def __repr__(self):
...
+        swap_label_pairs (Optional[List[Tuple[int, int]]]): The labels of
+            some datasets need to be aligned after image and label flipping.
+            For example, the segmentation labels left hand and right hand
+            in the human parsing task in natural images need to be converted
+            after the images are flipped left and right.
+    """
+
+    def __init__(self,
+                 prob: float,
+                 axes: Optional[Tuple[int, ...]] = None,
```
Comment (MeowZheng) [inline mmseg/datasets/transforms/transforms.py:1841]:
```suggestion
                 axes: Tuple[int, ...],
```
I think the axes cannot be None in this transform, and could you give an example about axes in docstring, like the order of axes is ZYX or ZXY?

File: mmseg/datasets/transforms/transforms.py [hunk 2]
```diff
@@ -1805,3 +1805,114 @@ def __repr__(self):
...
+    Apache-2.0 License Required Keys:
+
+    - img
+    - gt_seg_map (optional)
+    Modified Keys:
+    - img
+    - gt_seg_map (optional))
+    Added Keys:
+    - do_flip
+    - flip_axes
```
Comment (MeowZheng) [inline mmseg/datasets/transforms/transforms.py:1826]: Please other Bio3D transform and refine the docstring

File: mmseg/datasets/transforms/transforms.py (hunk 1 above)
Comment (Leeinsn) [reply mmseg/datasets/transforms/transforms.py]: Okay, the order 'ZXY' is given in docstring.

File: mmseg/datasets/transforms/transforms.py [hunk 3]
```diff
@@ -2015,3 +2015,116 @@ def __repr__(self):
...
+    https://github.com/MIC-DKFZ/batchgenerators/blob/master/batchgenerators/tra
+    nsforms/spatial_transforms.py # noqa:E501 Copyright 2021 Division of
+    Medical Image Computing, German Cancer Research Center (DKFZ) and Applied
+    Computer Vision Lab, Helmholtz Imaging Platform. Licensed under the
+    Apache-2.0 License.
+
+    Required Keys:
+
+    - img
+    - gt_seg_map (optional)
```
Comment (MeowZheng) [inline mmseg/datasets/transforms/transforms.py:2032]:
```suggestion

    Required Keys:
    - img (np.ndarry): Biomedical image with shape (N, Z, Y, X) by default,
        N is the number of modalities.
    - gt_seg_map (np.ndarray, optional): Biomedical seg map with shape
        (Z, Y, X) by default.
```

File: mmseg/datasets/transforms/transforms.py [hunk 4]
```diff
@@ -2015,3 +2015,116 @@ def __repr__(self):
...
+
+    Modified Keys:
+
+    - img
+    - gt_seg_map (optional)
```
Comment (MeowZheng) [inline mmseg/datasets/transforms/transforms.py:2037]:
```suggestion

    Required Keys:
    - img (np.ndarry): Biomedical image with shape (N, Z, Y, X) by default,
        N is the number of modalities.
    - gt_seg_map (np.ndarray, optional): Biomedical seg map with shape
        (Z, Y, X) by default.
```

File: mmseg/datasets/transforms/transforms.py [hunk 5]
```diff
@@ -2015,3 +2015,116 @@ def __repr__(self):
...
+
+
+@TRANSFORMS.register_module()
+class BioMedical3DRandomFlip(BaseTransform):
+    """Flip biomedical 3D images and segmentations. Modified from
+    https://github.com/MIC-DKFZ/batchgenerators/blob/master/batchgenerators/tra
+    nsforms/spatial_transforms.py # noqa:E501 Copyright 2021 Division of
+    Medical Image Computing, German Cancer Research Center (DKFZ) and Applied
+    Computer Vision Lab, Helmholtz Imaging Platform. Licensed under the
+    Apache-2.0 License.
```
Comment (MeowZheng) [inline mmseg/datasets/transforms/transforms.py:2027]:
```suggestion
    """Flip biomedical 3D images and segmentations.
    
    Modified from
    https://github.com/MIC-DKFZ/batchgenerators/blob/master/batchgenerators/transforms/spatial_transforms.py # noqa:E501
    Copyright 2021 Division of 
    Medical Image Computing, German Cancer Research Center (DKFZ) and Applied
    Computer Vision Lab, Helmholtz Imaging Platform.
    Licensed under the Apache-2.0 License.
```

**Review by MeowZheng — COMMENTED** [review]
why add a README file

**Review by MeowZheng — APPROVED** [review]

**Pull request merged**